src/bin/pgcopydb/ld_transform.c

```diff
@@ -1170,20 +1185,137 @@ parseMessage(StreamContext *privateContext, char *message, JSON_Value *json)
 
 
 /*
- * streamLogicalTransactionAppendStatement appends a statement to the current
- * transaction.
- *
- * There are two ways to append a statement to an existing transaction:
- *
- *  1. it's a new statement altogether, we just append to the linked-list
+ * coalesceLogicalTransactionStatement appends a new entry to an existing tuple
+ * array created during the last INSERT statement in a logical transaction.
  *
- *  2. it's the same statement as the previous one, we only add an entry to the
- *     already existing tuple array created on the previous statement
+ * This functionality enables the generation of multi-values INSERT or COPY
+ * commands, enhancing efficiency.
  *
- * This allows to then generate multi-values insert commands, for instance.
+ * Important: Before invoking this function, ensure that validation is performed
+ * using canCoalesceLogicalTransactionStatement.
```

Comment:
Makes me wonder if we should inline the whole thing within `streamLogicalTransactionAppendStatement` somehow. It would need to make sense that we do that for UDPATE and DELETE statements too, then, maybe, so that might be for another PR.

Reply:
>It would need to make sense that we do that for UDPATE and DELETE statements too, then, maybe, so that might be for another PR.

yeah, it was my thought process too.